ros/lib/utils.py

```diff
@@ -146,3 +152,21 @@ def calculate_percentage(numerator, denominator):
         return round((numerator / denominator) * 100, 2)
     else:
         return 0
+
+
+class MonitoringHandler(BaseHTTPRequestHandler):
```

Comment:
@patilsuraj767 Can we have a sad and happy path unittest for this class with an arbitrary set of threads,
to avoid regression in future and to skip on testing changes locally?

Reply:
Test cases added.